sdk/python/kfp_tekton/compiler/_data_passing_rewriter.py

```diff
@@ -109,67 +105,65 @@ def fix_big_data_passing(
-                        # workflow.parameters.* placeholders are not supported,
-                        # but the DSL compiler does not produce those.
-                        template_input_to_parent_constant_arguments.setdefault(
-                            (task_template_name, task_input_name),
-                            set()).add(argument_value)
-                    else:
-                        raise AssertionError
-
-                pipeline_input_name = extract_tekton_input_parameter_name(
-                    argument_value)
-                if pipeline_input_name:
+    # Indexing task arguments
+    pipeline_tasks = pipeline_template.get(
+        "tasks", []) + pipeline_template.get('finally', [])
+
+    for task in pipeline_tasks:
+        task_template_name = task['name']
+        parameter_arguments = task['params']
+        for parameter_argument in parameter_arguments:
+            task_input_name = parameter_argument['name']
+            argument_value = parameter_argument['value']
+
+            argument_placeholder_parts = deconstruct_tekton_single_placeholder(
+                argument_value)
+            if not argument_placeholder_parts:  # Argument is considered to be constant string
+                template_input_to_parent_constant_arguments.setdefault(
+                    (task_template_name, task_input_name),
+                    set()).add(argument_value)
+            else:
+                placeholder_type = argument_placeholder_parts[0]
+                if placeholder_type not in ('params', 'outputs', 'tasks',
+                                            'steps', 'workflow', 'pod',
+                                            'item'):
+                    # Do not fail on Jinja or other double-curly-brace templates
+                    continue
+                if placeholder_type == 'params':
+                    pipeline_input_name = argument_placeholder_parts[1]
                     template_input_to_parent_pipeline_inputs.setdefault(
                         (task_template_name, task_input_name), set()).add(
-                            (pipeline_template_name, pipeline_input_name))
-                else:
+                            (pipelinerun_name, pipeline_input_name))
+                elif placeholder_type == 'tasks':
+                    upstream_task_name = argument_placeholder_parts[1]
+                    assert argument_placeholder_parts[2] == 'results'
+                    upstream_output_name = argument_placeholder_parts[3]
+                    # upstream_template_name = upstream_task_name
+                    template_input_to_parent_task_outputs.setdefault(
+                        (task_template_name, task_input_name), set()).add(
+                            (upstream_task_name, upstream_output_name))
+                elif placeholder_type == 'item' or placeholder_type == 'workflow' or placeholder_type == 'pod':
```

Comment:
nit-pick -- this condition can be shortened to:
```Python
    elif placeholder_type in ('item', 'workflow', 'pod')
```

Reply:
Addressed